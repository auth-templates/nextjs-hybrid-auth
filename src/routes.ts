export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}