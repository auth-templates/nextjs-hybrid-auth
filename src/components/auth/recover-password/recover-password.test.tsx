import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { PublicRoutes } from '../../routes';
import RecoverPassword from './recover-password';

describe("RecoverPassword", () => {
    it('should contain all elements', () => {
        render(
            <MemoryRouter>
                <RecoverPassword onSend={jest.fn()}/>
            </MemoryRouter>
        );
        const email = screen.getByLabelText('Email');
        expect(email).toBeInTheDocument();
        expect(email).toHaveAttribute('required');
        expect(screen.getByText(/or go back to/)).toBeInTheDocument();
        expect(screen.getByRole('link', { name: /Sign in/ })).toHaveAttribute('href', PublicRoutes.login);
        expect(screen.getByRole('button', { name: "Send reset password email" })).toBeInTheDocument();
    });

    test('onSend callback receives the correct arguments', async () => {
        const mockOnSend= jest.fn();
        render(        
            <MemoryRouter>
                <RecoverPassword onSend={mockOnSend}/>
            </MemoryRouter>
        );
        const email = screen.getByLabelText('Email');
        userEvent.type(email, "account@mail.com");

        const sendButton = screen.getByRole('button', { name: "Send reset password email"});
        await userEvent.click(sendButton);
        expect(mockOnSend).toBeCalledTimes(1);
        expect(mockOnSend).toBeCalledWith({email:"account@mail.com"});    
    });

    it('input email should have theme danger when email is invalid', async () => {
        render(        
            <MemoryRouter>
                <RecoverPassword status={{theme: 'error', lines:['Email is invalid'] }} onSend={jest.fn()}/>
            </MemoryRouter>
        );
        const email = screen.getByLabelText('Email');
        expect(email).toHaveClass('danger');
    });

    it('email input should have theme normal when email is invalid and email input got focused', async () => {
        render(        
            <MemoryRouter>
                <RecoverPassword status={{theme: 'error', lines:['Email is invalid']}} onSend={jest.fn()}/>
            </MemoryRouter>
        );
        const email = screen.getByLabelText('Email');
        expect(email).toHaveClass('danger');
        await userEvent.click(email);
        expect(email).toHaveClass('normal');
    });

    it('displays InexistentAccount component when email is invalid', async () => {
        render(
            <MemoryRouter>
                <RecoverPassword status={{theme: 'error', lines:['Email is invalid']}} onSend={jest.fn()}/>
            </MemoryRouter>
        );
        expect(await screen.findByText(/No account with the email/)).toBeInTheDocument();
    });

    it("renders correctly with default props", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <RecoverPassword onSend={jest.fn()}/>
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
