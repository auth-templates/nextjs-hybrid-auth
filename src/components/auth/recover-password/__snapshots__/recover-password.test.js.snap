// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoverPassword renders correctly with default props 1`] = `
<div
  className="recoverPassword"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="normal"
        id="email"
        name="email"
        onFocus={[Function]}
        required={true}
        type="email"
      />
    </div>
    <div
      className="actionsGroup"
    >
      <div
        className="subgroup"
      >
        <button
          className="normal sendButton"
          type="submit"
        >
          Send reset password email
        </button>
        <div>
          or go back to 
          <a
            className="link"
            href="/login"
            onClick={[Function]}
          >
             Sign in 
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;
